gap/semigroups/semieunit.gi

```diff
@@ -0,0 +1,721 @@
+  out := [];
+  V := DigraphVertexLabels(McAlisterTripleSemigroupSemilattice(S));
+  for g in McAlisterTripleSemigroupGroup(S) do
+    for A in V do
+      if (McAlisterTripleSemigroupAction(S)(A, Inverse(g)) in V) then
+        Add(out, MTSE(S, A, g));
+      fi;
+    od;
+  od;
+  SetMcAlisterTripleSemigroupElmList(S, out);
+  return out;
+end);
+
+InstallMethod(String, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local G, X, Y;
+  G := McAlisterTripleSemigroupGroup(S);
+  X := McAlisterTripleSemigroupPartialOrder(S);
+  Y := McAlisterTripleSemigroupSemilattice(S);
+  return Concatenation("McAlisterTripleSemigroup(", String(G), ", ",
+                       String(X), ", ", String(Y), ")");
+end);
+
+#TODO Linebreak hints
+
+InstallMethod(ViewString, "for a McAlister triple semigroup",
+[IsMcAlisterTripleSemigroup],
+function(S)
+  local G;
+  G := McAlisterTripleSemigroupGroup(S);
+  return Concatenation("<McAlister triple semigroup over ", ViewString(G), ">");
+end);
+
+# Currently this does not work. iso_x restricted to the semilattice is not
+# necesarilly an insomorphism of the two semilattices. If we have an isomorphism
+# of the semilattices then not sure how to extend to the full isomorphism.
+
+InstallMethod(IsomorphismSemigroups, "for two McAlister triple semigroups",
+[IsMcAlisterTripleSemigroup, IsMcAlisterTripleSemigroup],
+function(S, T)
+  local YS, YT, XT, iso_g, iso_x, im_YS, rep, A;
+
+  iso_g := IsomorphismGroups(McAlisterTripleSemigroupGroup(S),
+                             McAlisterTripleSemigroupGroup(T));
+
+  if iso_g = fail then
+    return fail;
+  fi;
+
+  YS := McAlisterTripleSemigroupSemilattice(S);
+  YT := McAlisterTripleSemigroupSemilattice(T);
+  XT := McAlisterTripleSemigroupPartialOrder(T);
+
+  if not IsIsomorphicDigraph(YS, YT) then
+    return fail;
+  fi;
+
+  iso_x := IsomorphismDigraphs(McAlisterTripleSemigroupPartialOrder(S), XT);
+
+  if iso_x = fail then
+    return fail;
+  fi;
+
+  im_YS := List(DigraphVertexLabels(YS), a -> a ^ iso_x);
+  # if the restriction of iso_x to DigraphVertexLabels(YS) is not
+  # DigraphVertexLabels(YT) then we need to compose iso_x with an
+  # automorphism of McAlisterTripleSemilattice(T). Composing this with
+  # iso_x will restrict to an isomorphism from (the labels of) YS to YT.
+  if not im_YS = DigraphVertexLabels(YT) then
+    A := AutomorphismGroup(XT);
+    rep := RepresentativeAction(A, im_YS, DigraphVertexLabels(YT), OnSets);
+  else
+    rep := ();
+  fi;
+
+  if ForAll(McAlisterTripleSemigroupGroup(S),
+      g -> ForAll(DigraphVertices(McAlisterTripleSemigroupPartialOrder(S)),
+      x -> (McAlisterTripleSemigroupAction(S)(x, g)) ^ (rep * iso_x) =
+      McAlisterTripleSemigroupAction(T)((x ^ iso_x), (g ^ iso_g)) ^ rep)) then
+    return MappingByFunction(S, T, s -> MTSE(T, s[1] ^ iso_x, s[2] ^ iso_g));
+  fi;
+
+  return fail;
+end);
+
+InstallMethod(IsIsomorphicSemigroup, "for two McAlister triple semigroups",
```

Comment:
You can get rid of this method: the one in `gap/attributes/isomorph.gi` works perfectly fine for MTSs.